### PR TITLE
Implement light coloring

### PR DESCRIPTION
**Pull request by Victormeriqui**
- Adds colored lights with correct blending/mixing

Replaces previous intensity only functions with RGB return values that represent the light's color blended with it's intensity at the specified point

image
image

### CODE REVIEW

File: src/Engine/Rendering/Lighting/DirectionalLight.cpp [hunk 1]
```diff
@@ -16,27 +16,10 @@ namespace Engine
...
-				depthbuffer(DepthBuffer(200, 200)),
-				projection_mat(Matrix4().SetOrthographicProjection(-ortho_size, ortho_size, ortho_size, -ortho_size, -ortho_size, ortho_size))
-			{
-				SetDirection(direction);
-			}
 
-			DirectionalLight::DirectionalLight(const Vector3& direction, float intensity) :
+			DirectionalLight::DirectionalLight(const Vector3& direction, float intensity, RGBColor color) :
 				intensity(intensity),
+				color(RGBColor()),
```
Comment (AshleighAdams) [inline src/Engine/Rendering/Lighting/DirectionalLight.cpp:22]: Why isn't intensity just the length of the color?

**Review by AshleighAdams — APPROVED** [review]
I still think lights should just be a part of the shaders only, and you set the lights in the shader, but otherwise lgtm 

I know it's part of a "shaded shader" but I wonder if some multi pass thing could be done instead, then again I'm reviewing on my phone so I might be missing lots

File: src/Engine/Rendering/Lighting/DirectionalLight.cpp (hunk 1 above)
Comment (Victormeriqui) [reply src/Engine/Rendering/Lighting/DirectionalLight.cpp:22]: Cause I didn't think about that, nice idea 👀 :D

Comment (Victormeriqui) [reply src/Engine/Rendering/Lighting/DirectionalLight.cpp:22]: And regarding the first comment, right now I really dislike how light transformations space are done in the shader along with how the light depth buffer is passed for shadow maps

I think it's a problem that needs to be fixed eventually, but I still like having a shader independent lighting system, where you can add and remove lights, set ambient lighting etc...IMO it's up to the the shader how it wants to allow the lights to influence it's results (if at all), but doesn't have to manage the lights or hold them

**Pull request merged**